File: go/sort/sortUtil.go
package sort

import "math/rand"

func Swap[T any](arr []T, i, j int) {
	if i == j {
		return
	}
	arr[i], arr[j] = arr[j], arr[i]
}

func RandomArray(size int, min int, max int) []int {
	arr := make([]int, size)
	for i := 0; i < size; i++ {
		arr[i] = rand.Intn(max-min) + min
	}
	return arr
}

func CompareAsc(a, b int) bool {
	return a > b
}

func CompareDesc(a, b int) bool {
	return a < b
}

// Check if array is sorted in ascending order
func IsAscArray(arr []int) bool {
	for i := 0; i < len(arr)-1; i++ {
		if arr[i] > arr[i+1] {
			return false
		}
	}
	return true
}

// Check if array is sorted in descending order
func IsDescArray(arr []int) bool {
	for i := 0; i < len(arr)-1; i++ {
		if arr[i] < arr[i+1] {
			return false
		}
	}
	return true
}

File: dart/src/sort/sortUtil.dart
import 'dart:math';

/**
 * Generate random list
 * @param length 长度
 * @param max 范围 [1,max]
 */
List<int> randomList(int length, int max) {
  List<int> list = [];
  for (int i = 0; i < length; i++) {
    list.add(Random().nextInt(max) + 1);
  }
  return list;
}

/**
 * 是否是升序数组
 */
bool isAscArray(List list) {
  for (int i = 0; i < list.length - 1; i++) {
    if (list[i] > list[i + 1]) {
      return false;
    }
  }
  return true;
}

/**
 * 是否是降序数组
 */
bool isDescArray(List list) {
  for (int i = 0; i < list.length - 1; i++) {
    if (list[i] < list[i + 1]) {
      return false;
    }
  }
  return true;
}

void swap(List list, int i, int j) {
  int temp = list[i];
  list[i] = list[j];
  list[j] = temp;
}
